v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/ddl/Ddl.java

```diff
@@ -98,7 +99,8 @@ public List<String> tablesReferenced(String tableName) {
       throw new IllegalStateException(
           "attempting to fetch table which does not exist in spanner:" + tableName);
     }
-    if (table.interleaveInParent() != null) {
+    if (table.interleaveInParent() != null
+        && !Objects.equals(table.interleaveType(), new String("IN"))) {
```

Comment:
we need to support both types of interleaving here.

Reply:
This ensures that if it is INTERLEAVE IN then dont a dd a parent, else add a parent. Support for both types will work